### PR TITLE
doc: add controller env variable examples

### PR DESCRIPTION
**Pull request by saumeya**
Signed-off-by: saumeya <saumeyakatyal@gmail.com>

**What type of PR is this?**

 /kind documentation

**What does this PR do / why we need it**:

**Have you updated the necessary documentation?**

* [x] Documentation update is required by this PR.
* [x] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #?

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: docs/reference/argocd.md [hunk 1]
```diff
@@ -156,6 +156,42 @@ spec:
...
+  labels:
+    example: controller
+spec:
+  controller:
+    env:
+    - name: ARGOCD_APPLICATION_CONTROLLER_REPO_SERVER_TIMEOUT_SECONDS
+      valueFrom:
+      configMapKeyRef:
+        key: controller.repo.server.timeout.seconds
+        name: argocd-cmd-params-cm
```
Comment (jannfis) [inline docs/reference/argocd.md:193]: Tiny nit: I think this should be intended one level deeper.

Comment (saumeya) [reply docs/reference/argocd.md:193]: You are correct, will update it

File: docs/reference/argocd.md [hunk 2]
```diff
@@ -156,6 +156,42 @@ spec:
...
+    example: controller
+spec:
+  controller:
+    env:
+    - name: ARGOCD_APPLICATION_CONTROLLER_REPO_SERVER_TIMEOUT_SECONDS
+      valueFrom:
+        configMapKeyRef:
+          key: controller.repo.server.timeout.seconds
+          name: argocd-cmd-params-cm     
+```
```
Comment (jannfis) [inline docs/reference/argocd.md:194]: I think this method does not make sense in the context of the Operator. We should not advocate for it. Can you please remove it?

Comment (saumeya) [reply docs/reference/argocd.md:194]: Sure

**Pull request merged**